src/electron-main.js

```diff
@@ -696,7 +696,17 @@ app.on('ready', async () => {
             spellcheck: true,
         },
     });
-    mainWindow.loadURL('vector://vector/webapp/');
+
+    if (vectorConfig['proxy']) {
+      console.log(`Starting electron through proxy: ${vectorConfig['proxy']}`);
+      mainWindow.webContents.session.setProxy({proxyRules: vectorConfig['proxy']})
```

Comment:
You should be able to use `await` here, and then you can remove the `then` handler and put `loadURL` back where it was outside of the `if` to cover both cases.